ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/CaseEvent.java

```diff
@@ -41,4 +46,10 @@
     public String getValue() {
         return value;
     }
+
+    public static CaseEvent fromValue(String value) {
+        return Arrays.stream(values()).filter(event -> event.value.equals(value))
+            .findFirst()
+            .orElseThrow(() -> new BadRequestException("Unknown Case Event: " + value));
```

Comment:
Not sure "Bad Request" is semantically appropriate. At this level, perhaps `IllegalArgumentException` is better?

Reply:
True! Was just trying to mimick the logic which was already there - let's see how it looks like with a lower level exception